app/src/androidTest/java/com/github/h3lp3rs/h3lp/MedicalCardActivityTest.kt

```diff
@@ -1,126 +1,220 @@
 package com.github.h3lp3rs.h3lp
+
 import android.app.Activity
-import android.app.Instrumentation
+import android.app.Instrumentation.*
 import android.content.Context
 import android.content.Intent
 import android.view.View
-import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ActivityScenario.*
+import androidx.test.core.app.ApplicationProvider.*
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions.*
 import androidx.test.espresso.assertion.ViewAssertions.matches
-import androidx.test.espresso.intent.Intents
-import androidx.test.espresso.intent.matcher.IntentMatchers
+import androidx.test.espresso.intent.Intents.*
+import androidx.test.espresso.intent.matcher.IntentMatchers.*
 import androidx.test.espresso.matcher.RootMatchers.isDialog
 import androidx.test.espresso.matcher.ViewMatchers.*
-import androidx.test.ext.junit.rules.ActivityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.h3lp3rs.h3lp.database.Databases.*
+import com.github.h3lp3rs.h3lp.database.MockDatabase
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.globalContext
+import com.github.h3lp3rs.h3lp.signin.SignInActivity.Companion.userUid
+import com.github.h3lp3rs.h3lp.storage.Storages
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.resetStorage
+import com.github.h3lp3rs.h3lp.storage.Storages.Companion.storageOf
 import com.google.android.material.textfield.TextInputLayout
 import org.hamcrest.Description
 import org.hamcrest.Matcher
 import org.hamcrest.Matchers
 import org.hamcrest.Matchers.not
 import org.hamcrest.TypeSafeMatcher
-import org.junit.Rule
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import java.util.*
+
 @RunWith(AndroidJUnit4::class)
 class MedicalCardActivityTest {
-    private val ctx: Context = ApplicationProvider.getApplicationContext()
-    @get:Rule
-    val testRule = ActivityScenarioRule(
-        MedicalCardActivity::class.java
-    )
+    private val ctx: Context = getApplicationContext()
+
+    private fun launch(): ActivityScenario<MedicalCardActivity> {
+        return launch(Intent(getApplicationContext(), MedicalCardActivity::class.java))
+    }
+
+    private fun launchAndDo(action: () -> Unit) {
+        launch().use {
+            start()
+            action()
+            end()
+        }
+    }
+
+    private fun start() {
+        init()
+        val intent = Intent()
+        val intentResult = ActivityResult(Activity.RESULT_OK, intent)
+        intending(anyIntent()).respondWith(intentResult)
+    }
+
+    private fun end() {
+        release()
```

Comment:
Dont know if i really like this use of release() and init() without Intents.
Eventhough it is smaller I think we lose a bit in our code clarity not knowing what we are initiating / releasing

Reply:
It was given as piece of advice on the launchpad (week 1). We all know what it means, and the code is boilerplate either way. So I think keeping it as short as possible is cleaner.